arrows/core/transform_detected_object_set.h

```diff
@@ -0,0 +1,114 @@
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef KWIVER_ARROWS_TRANSFORM_DETECTED_OBJECT_SET_H_
+#define KWIVER_ARROWS_TRANSFORM_DETECTED_OBJECT_SET_H_
+
+#include <arrows/core/kwiver_algo_core_export.h>
+
+#include <vital/algo/detected_object_filter.h>
+#include <vital/io/camera_io.h>
+
+using namespace kwiver::vital;
+
+namespace kwiver {
+namespace arrows {
+namespace core {
+
+/// Transforms detections based on source and destination cameras.
+class KWIVER_ALGO_CORE_EXPORT transform_detected_object_set
+  : public vital::algorithm_impl<transform_detected_object_set,
+                                 vital::algo::detected_object_filter>
+{
+public:
+  PLUGIN_INFO( "transform_detected_object_set",
+               "Transforms a detected object set based on source and "
+               "destination cameras.\n\n" )
+
+  /// Default constructor
+  transform_detected_object_set();
+
+  /// Constructor taking source and destination cameras directly
+  transform_detected_object_set(kwiver::vital::camera_perspective_sptr src_cam,
+                                kwiver::vital::camera_perspective_sptr dest_cam);
+
+  /// Default destructor
+  virtual ~transform_detected_object_set() = default;
+
+  /// Get this algorithm's configuration block
+  virtual vital::config_block_sptr get_configuration() const;
+
+  /// Set this algorithm's properties via a config block
+  virtual void set_configuration(vital::config_block_sptr config);
+
+  /// Check that the algorithm's currently configuration is valid
+  virtual bool check_configuration(vital::config_block_sptr config) const;
+
+  /// Apply the transformation
+  virtual vital::detected_object_set_sptr
+    filter( const vital::detected_object_set_sptr input_set) const;
+
+private:
+  std::string src_camera_krtd_file_name;
+  std::string dest_camera_krtd_file_name;
+
+  kwiver::vital::camera_perspective_sptr src_camera;
+  kwiver::vital::camera_perspective_sptr dest_camera;
+
+  virtual vital::bounding_box<double>
+    transform_bounding_box(vital::bounding_box<double>& bbox) const;
```

Comment:
The input here should be `vital::bounding_box<double>const& bbox`, note the `const`.  This is because you are not changing the memory of `bbox` that is passed in.  Instead you are returning a new object with the transformed values.  An alternative approach is to modify the memory in place by passing `bbox` as you are, modifying that object in place, and then returning `void` or maybe `bool` (if the function could fail).  I think the best option here is to pass by `const&`.

Reply:
Understood, fixed!